aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java

```diff
@@ -1333,13 +1353,19 @@ public void run() {
                         ((NewPasswordContinuation) signInChallengeContinuation)
                                 .setPassword(signInChallengeResponse);
                         signInChallengeContinuation.setClientMetaData(clientMetadata);
+                        for (final String key : userAttributes.keySet()) {
+                            signInChallengeContinuation.setChallengeResponse(CHALLENGE_RESPONSE_USER_ATTRIBUTES_PREFIX_KEY + key, userAttributes.get(key));
+                        }
                         detectedContinuation = signInChallengeContinuation;
                         signInCallback = new InternalCallback<SignInResult>(callback);
                         break;
                     case CUSTOM_CHALLENGE:
                         signInChallengeContinuation.setChallengeResponse("ANSWER", signInChallengeResponse);
                         detectedContinuation = signInChallengeContinuation;
                         signInCallback = new InternalCallback<SignInResult>(callback);
+                        for (final String key : userAttributes.keySet()) {
+                            signInChallengeContinuation.setChallengeResponse(CHALLENGE_RESPONSE_USER_ATTRIBUTES_PREFIX_KEY + key, userAttributes.get(key));
+                        }
```

Comment:
[iOS implementation of mobile client](https://github.com/aws-amplify/aws-sdk-ios/blob/2ce94a7fda175ed6ffd86c993b0ace219b926ac2/AWSAuthSDK/Sources/AWSMobileClient/AWSMobileClientExtensions.swift#L685) does not add user attributes to custom challenge continuation. Is this required?

Reply:
Removing this.